readthedocs/api/v2/mixins.py

```diff
@@ -35,15 +35,16 @@ def _get_cache_tags(self):
            so any exceptions like 404 should be caught.
         """
         try:
+            tags = []
             project = self._get_project()
             version = self._get_version()
-            tags = [
-                project.slug,
-                get_cache_tag(project.slug, version.slug),
-            ]
-            if self.project_cache_tag:
+            if project:
+                tags.append(project.slug)
+            if project and version:
+                tags.append(get_cache_tag(project.slug, version.slug))
+            if project and self.project_cache_tag:
                 tags.append(get_cache_tag(project.slug, self.project_cache_tag))
             return tags
-        except Exception as e:
+        except Exception:
             log.debug('Error while retrieving project and version for this view.')
```

Comment:
Can we scope this exception down at all? Or should we at least log a warning? Blanket exceptions with nothing in Sentry scare me. 

Guessing we might not need it any more, since we made the code more defensive, but step 1 is a `log.warning` or `log.exception` :)